### PR TITLE
Cat display

### PR DESCRIPTION
**Pull request by MaximilianMeiler**
Currently displays cat pelt (+x/y), hats, eyes from db.
Cat layout isn't quite what I want yet, so don't treat it as final - just wanted to get something functional on the repo

### CODE REVIEW

File: Frontend/src/App.jsx [hunk 1]
```diff
@@ -3,15 +3,20 @@ import axios from 'axios'
...
 import Login from "./Login"
 // import Rewards from "./Rewards"
+import Cat from "./Cat"
+
 
 axios.defaults.headers.post['Content-Type'] = 'application/json'; //is this needed in this file?
 
 function App() {
   const [courses, setCourses] = useState(null);
+  // eslint-disable-next-line no-unused-vars
```
Comment (kthchew) [inline Frontend/src/App.jsx:13]: image

File: Frontend/src/App.jsx [hunk 2]
```diff
@@ -3,15 +3,20 @@ import axios from 'axios'
...
   const [userId, setUserId] = useState(null); //canvas user id
   const [userData, setUserData] = useState(null); //from db
   const [apiKey, setApiKey] = useState(null)
   const [overlay, setOverlay] = useState("login")
+  const [width, setWidth] = useState(null);
+  const [height, setHeight] = useState(null)
```
[the remark this answers is not in the document]
Comment (MaximilianMeiler) [reply Frontend/src/App.jsx:19]: might as well

File: Frontend/src/App.jsx [hunk 3]
```diff
@@ -48,17 +53,56 @@ COURSE STORAGE - NEW MODEL
...
+  useEffect(() => {
+    const setDimensions = () => {
+      setWidth(window.innerWidth);
+      setHeight(window.innerHeight);
+    }
+
+    setDimensions()
+
+    window.addEventListener('resize', setDimensions)
+  }, [])
```
[the remark this answers is not in the document]
Comment (MaximilianMeiler) [reply Frontend/src/App.jsx:65]: seems legit

File: Frontend/src/App.jsx [hunk 4]
```diff
@@ -48,17 +53,56 @@ COURSE STORAGE - NEW MODEL
...
+              //note - we may want to make the page refresh on window resize
+              const deskWidth = 132;
+              const deskHeight = 96;
+
+              var desksPerRow = Math.floor((width * .86) / deskWidth)
+              var desksPerCol = (Math.ceil(userData.cats.length / desksPerRow));
+              
+              var yCoord = deskHeight + height * .75 * (Math.floor(i / desksPerRow) / desksPerCol);
+              var xCoord = (i % desksPerRow) * deskWidth
+              var offset = ((width / height) * (yCoord - deskHeight)) % deskWidth - deskWidth
```
[the remark this answers is not in the document]
Comment (MaximilianMeiler) [reply Frontend/src/App.jsx:92]: it doesn't really (yet)

**Review by kthchew — APPROVED** [review]

**Pull request merged**